lib/IRGen/GenEnum.cpp

```diff
@@ -659,7 +659,8 @@ namespace {
     void consume(IRGenFunction &IGF, Explosion &src,
                  Atomicity atomicity,
                  SILType T) const override {
-      if (tryEmitConsumeUsingDeinit(IGF, src, T)) {
+      if (tryEmitConsumeUsingDeinit(IGF, src, T,
+                                    false /*hasMetadataUnaccessibleNonMovableField*/)) {
```

Comment:
Can we compute hasMetadataUnaccessibleNonMovableField inside of tryEmitConsumeUsingDeinit, instead of passing it in as a flag?

Reply:
I compute that property in the same place where we compute similar layout information in Enum and Struct TypeInfo computation and thought this to be preferable since we only have to do it once.

e.g here https://github.com/swiftlang/swift/pull/75967/files#diff-e106653d30372f2fcea6dad4bb23fce68d058bfc3508e5bc5fb24c57c186cf4fR968
and here
https://github.com/swiftlang/swift/pull/75967/files#diff-c3cfd9d79c4c326e449e27fe94c8358efada8e5604b951d9a6bc271542f263a5R6490

But yes, I can also do something similar as getSingletonAggregateFieldType in `tryEmitDeinitCall` (i.e look for enum and struct and then go through its fields).

Reply:
Hmm, I am thinking instead of passing this flag as an argument I can expose it as a predicate on TypeInfo.